File: Creational/Builder/PHP/builders/Director.php
<?php

require_once(__DIR__.'/../interfaces/IBuilder.php');
require_once(__DIR__.'/../products/EngineV8.php');

/**
 * Director builder.
 */

class Director
{
	public function constructSportCar(IBuilder &$builder)
	{
		$builder->reset();
		$builder->setEngine(new EngineV8(4.2));
		$builder->setSeats(2);
		$builder->setTripComputer(false);
	}
}

File: Creational/Builder/C++/builders/Director.cpp
#include "Director.h"
#include "./../interfaces/IBuilder.h"
#include "./../products/EngineV8.h"

void Director::constructSportCar(IBuilder* builder)
{
    builder->reset();
    builder->setEngine(new EngineV8(4.2));
    builder->setSeats(2);
    builder->setTripComputer(false);
}
